### ISSUE TITLE
Exception in chuncked response

### ISSUE THREAD

**Issue by riksking** [dotnet/runtime#68581]
Hi there.

**Describe the bug**
I am migrating a project from .Net Framework 4.7.2 to .NET 6.
And I am getting error in response when it is serialized.

`System.ServiceModel.ProtocolException: There is a problem with the XML data received from the network. See the description of the inner exception for details.
 ---> System.Xml.XmlException: The data at the root level is invalid. Line 1, position 1.
`

I think this is due to the fact that the response does not support chunking.

I got response like this

```
20d
<soap:Envelope xmlns:soap="http://schemas.xmlsoap.org/soap/envelope/"><soap:Body><ns2:searchClientRs xmlns:ns2="........">
    <messageHeader>
        <messageId>689b1900-ef74-4352-90fd-d180d59c1a37</messageId>
        <correlationId>689b1900-ef74-4352-90fd-d180d59c1a37</correlationId>
        <responseStatus>
            <code>0</code>
            <textMessage/>
            <warning xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:nil="true"/>
        </responseStatus>
    </messageHead
33f
er>
    <clients>
        <client>
            <bankId>1</bankId>
...
```

Symbols "**20d**" and "**33f**" it's special symbol in chunked response, as far as I understand))

I tried using the TransferMode setting to enable chunking.
But neither the TransferMode.Stream setting nor the TransferMode.StreamResponse setting helped and returned the same error.

I found this code in source 

```
namespace System.ServiceModel.Channels
{
    internal class HttpResponseMessageHelper
    {
    ...
        private Task<Message> ReadStreamAsMessageAsync(TimeoutHelper timeoutHelper)
        {
            var content = _httpResponseMessage.Content;
            Task<Stream> contentStreamTask = GetStreamAsync();
 
            if (TransferModeHelper.IsResponseStreamed(_factory.TransferMode))
            {
                return ReadStreamedMessageAsync(contentStreamTask);
            }
            if (!content.Headers.ContentLength.HasValue)
            {
                return ReadChunkedBufferedMessageAsync(contentStreamTask, timeoutHelper);
            }
            return ReadBufferedMessageAsync(contentStreamTask);
        }
```

And I checked all three read methods. 
They all throw the same exception.

**To Reproduce**
I'm working with a SOAP service that I don't have access to and it's written in Java.
So I don't know how to reproduce this code.
I can attach the full text of the response with headers.

**Expected behavior**
I expect that everything will work just as well as on the .NET Framework 4.7.2.

**Additional context**
Soap UI correctly parses this response.

![image](https://user-images.githubusercontent.com/13674964/165062754-70bfbcd0-f6e4-4921-9df4-e732906715dd.png)

**Comment by MihaZupan** [comment]
You are correct that this is a bug and we should give priority to `Transfer-Encoding` over `Content-Length`.
The exact issue has been mentioned in the past in https://github.com/dotnet/runtime/issues/63053#issuecomment-999033021 and https://github.com/dotnet/runtime/issues/62906#issuecomment-996078804.
Not closing this as a duplicate yet so we can retriage priority (7.0 vs future).

**Comment by mconnew** [comment]
I would like to call out that there is no workaround for this bug. The person who reported the issue doesn't own the service, so this is a blocker to being able to adopt .NET Core. The server in question is the Java Jetty server which is a mainstream and popular server and not some obscure unusual implementation.

**Comment by wfurt** [comment]
should we reopen this for servicing @karelz since there is no workaround and this violates RFC?
(if so, it may be worth of validating fix with daily builds)

**Comment by mconnew** [comment]
I vote to back port this to .NET 6 as it's the LTS version.

**Comment by MihaZupan** [comment]
Reopening for backporting consideration.

**Comment by pedrobsaila** [comment]
#69085 should also be backported with it, because the test code in this breaks with NodeJs

**Comment by antonfirsov** [comment]
Triage: blocking compat issue, we should backport the fix to 6.0. The only risk is if someone is relying on the current non RFC-compliant 6.0 behavior.

**Comment by MihaZupan** [comment]
Fixed for 7.0 in #69016 and backported to 6.0.8 in #70744.
